### PR TITLE
Homework make_shared(optional)

### PR DESCRIPTION
**Pull request by J-Cieplinski**
make_shared additional homework

Authors:
@J-Cieplinski
@majkel84

### CODE REVIEW

**Review by MaciejSalamonski — APPROVED** [review]
Just wow. Nice code!

**Review by jakjan95 — APPROVED** [review]
For me coode looks good, I approve it

File: homework/shared_ptr/shared_ptr.hpp [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    void swap(shared_ptr<T>& secondPointer) noexcept;
+    const T* get() const;
+    void reset(T* newPtr = nullptr);
+
+    const T* operator->();
+    T& operator*();
+    explicit operator bool() const noexcept;
+    shared_ptr<T>& operator=(const shared_ptr<T>& ptr) noexcept;             //copy assignment
+    shared_ptr<T>& operator=(shared_ptr<T>&& previousOwner);  //move assignment
+
```
Comment (WojtekMs) [inline homework/shared_ptr/shared_ptr.hpp:30]: I think you forgot about noexcept here ;)

**Review by WojtekMs — COMMENTED** [review]
Good work :)

**Pull request closed without merging**